app/uk/gov/hmrc/exports/services/reversemapping/declaration/XmlParsingException.scala

```diff
@@ -16,8 +16,4 @@
 
 package uk.gov.hmrc.exports.services.reversemapping.declaration
 
-import scala.xml.NodeSeq
-
-trait XmlParser[A] {
-  def parse(inputXml: NodeSeq): A
-}
+case class XmlParsingException(message: String) extends RuntimeException(message)
```

Comment:
Might be nice to add the xml that the exception was thrown when parsing

Reply:
But this will be visible in the logs and I am not sure if we can show the whole xml there.

Reply:
Well maybe we can just add the section of the xml that failed? Would be useful IMO

Reply:
I simply don't know, but we can pass the question to BAs.